app/models/host.rb

```diff
@@ -225,30 +226,6 @@ def raise_cluster_event(ems_cluster, event)
   end
   private :raise_cluster_event
 
-  # is_available?
-  # Returns:  true or false
-  #
-  # The UI calls this method to determine if a feature is supported for this Host
-  # and determines if a button should be displayed.  This method should return true
-  # even if a function is not 'currently' available due to some condition that is not
-  # being met.
-  def is_available?(request_type)
-    send("validate_#{request_type}")[:available]
-  end
-
-  # is_available_now_error_message
-  # Returns an error message string if there is an error.
-  # Returns nil to indicate no errors.
-  # This method is used by the UI along with the is_available? methods.
-  def is_available_now_error_message(request_type)
-    send("validate_#{request_type}")[:message]
-  end
-
-  def raise_is_available_now_error_message(request_type)
```

Comment:
Yes, that could be done too, except that this method raises `MiqException::MiqVmError`
exception (which is then caught somewhere in controller code), i.e. a VM specific
error and I don't have a very good sense on how to make this mechanism
generic / pluggable.